transformer/events_creator.go

```diff
@@ -32,22 +32,30 @@ import (
 )
 
 type Transformer struct {
-	context       context.Context
-	eval          evaluator.Evaluator
-	eventMetadata common.MapStr
-	events        []beat.Event
+	context       	   context.Context
+	eval          	   evaluator.Evaluator
+	eventMetadata 	   common.MapStr
+	events        	   []beat.Event
+	commonData 		   CommonDataInterface
 }
 
-func NewTransformer(ctx context.Context, eval evaluator.Evaluator, index string) Transformer {
+func NewTransformer(ctx context.Context, eval evaluator.Evaluator, cdProvider CommonDataProvider, index string) (Transformer, error) {
```

Comment:
Sorry for changing my mind, but now seeing that:
1. the cdp is only used once
2. this changes `NewTransformer` which is sort of a constructor to a possibly failing function

maybe we pass the common data directly? and use the provider in `beater/cloudbeat.go > New`